engines/efh/files.cpp

```diff
@@ -0,0 +1,378 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "efh/efh.h"
+
+namespace Efh {
+
+int32 EfhEngine::readFileToBuffer(Common::String &filename, uint8 *destBuffer) {
+	debugC(1, kDebugUtils, "readFileToBuffer %s", filename.c_str());
+	Common::File f;
+	if (!f.open(filename))
+		error("Unable to find file %s", filename.c_str());
+
+	int size = f.size();
+
+	return f.read(destBuffer, size);
+}
+
+void EfhEngine::readAnimInfo() {
+	debugC(6, kDebugEngine, "readAnimInfo");
+
+	Common::String fileName = "animinfo";
+	Common::File f;
+	if (!f.open(fileName))
+		error("Unable to find file %s", fileName.c_str());
+
+	for (int i = 0; i < 100; ++i) {
+		for (int id = 0; id < 15; ++id) {
+			_animInfo[i]._unkAnimArray[id]._field[0] = f.readByte();
+			_animInfo[i]._unkAnimArray[id]._field[1] = f.readByte();
+			_animInfo[i]._unkAnimArray[id]._field[2] = f.readByte();
+			_animInfo[i]._unkAnimArray[id]._field[3] = f.readByte();
+
+			debugC(6, kDebugEngine, "%d %d %d %d", _animInfo[i]._unkAnimArray[id]._field[0], _animInfo[i]._unkAnimArray[id]._field[1], _animInfo[i]._unkAnimArray[id]._field[2], _animInfo[i]._unkAnimArray[id]._field[3]);
+		}
+
+		Common::String debugStr = "";
+		for (int id = 0; id < 10; ++id) {
+			_animInfo[i]._field3C_startY[id] = f.readByte();
+			debugStr += Common::String::format("%d ", _animInfo[i]._field3C_startY[id]);
+		}
+		debugC(6, kDebugEngine, "%s", debugStr.c_str());
+
+		debugStr = "";
+		for (int id = 0; id < 10; ++id) {
+			_animInfo[i]._field46_startX[id] = f.readUint16LE();
+			debugStr += Common::String::format("%d ", _animInfo[i]._field46_startX[id]);
+		}
+		debugC(6, kDebugEngine, "%s", debugStr.c_str());
+		debugC(6, kDebugEngine, "---------");
+	}
+}
+
+void EfhEngine::findMapFile(int16 mapId) {
+	debugC(7, kDebugEngine, "findMapFile %d", mapId);
+
+	if (!_introDoneFl)
+		return;
+
+	Common::String fileName = Common::String::format("map.%d", mapId);
+	Common::File f;
+	// The original was checking for the file and eventually asking to change floppies
+	if (!f.open(fileName))
+		error("File not found: %s", fileName.c_str());
+
+	f.close();
+}
+
+void EfhEngine::rImageFile(Common::String filename, uint8 *targetBuffer, uint8 **subFilesArray, uint8 *packedBuffer) {
+	debugC(1, kDebugUtils, "rImageFile %s", filename.c_str());
+	readFileToBuffer(filename, packedBuffer);
+
+#ifndef debug
+	uncompressBuffer(packedBuffer, targetBuffer);
+#else
```

Comment:
I'd recommend wrapping this into `if (ConfMan.getBool("dump_scripts"))` at run-time